### ISSUE TITLE
projectDir/bin/script not found 

### ISSUE THREAD

**Issue by chelauk** [nextflow-io/nextflow#2324]
## Bug report 

(singularity does not find script in $projectDir/bin folder)

### Expected behavior and actual behavior

singularity should bind to $projectDir/bin and run

### Steps to reproduce the problem

(Provide a test case that reproduce the problem either with a self-contained script or GitHub repository)

### Program output 

Sep-16 12:13:51.930 [Task monitor] ERROR nextflow.processor.TaskProcessor - Error executing process > 'NFCORE_TEST:TEST:INPUT_CHECK:SAMPLESHEET_CHECK (samplesheet_test_illumina_amplicon.csv)'

Caused by:
  Process `NFCORE_TEST:TEST:INPUT_CHECK:SAMPLESHEET_CHECK (samplesheet_test_illumina_amplicon.csv)` terminated with an error exit status (127)

Command executed:

  check_samplesheet.py \
      samplesheet_test_illumina_amplicon.csv \
      samplesheet.valid.csv

Command exit status:
  127

Command output:
  (empty)

Command error:
  WARNING: Skipping mount /var/singularity/mnt/session/etc/resolv.conf [files]: /etc/resolv.conf doesn't exist in container
  .command.sh: line 3: check_samplesheet.py: command not found

Work dir:
  /data/scratch/DMP/UCEC/EVGENMOD/cjames/test/work/5b/f52306672e571377cf3bd0e8d27c9f

Tip: view the complete command output by changing to the process work dir and entering the command `cat .command.out`
Sep-16 12:13:51.957 [Task monitor] INFO  nextflow.Session - Execution cancelled -- Finishing pending tasks before exit

### Environment 

* Nextflow version: 21.04.3 build 5560
* Java version: 1.8.0_261
* Operating system: Linux
* Bash version: GNU bash, version 4.4.19(1)

### Additional context
I compared the .command.run scripts after successful vs failing runs and I realised that these lines were different:
1. failing:  `export PATH="/home/cjames/scratch/nextflow_pipelines/nf-core-test/bin:\$PATH"`
2. successful: `export PATH="/data/scratch/DMP/UCEC/EVGENMOD/cjames/nextflow_pipelines/nf-core-test/bin:\$PATH"`

 so when I navigate using the full path (case 2): success but when I use the "scratch" soft link as in case 1 it fails
 
 Is it possible ensure that nextflow sets the projectDir variable with the full path?

**Comment by pditommaso** [comment]
Nextflow mounts singularity host paths only when using the `autoMounts` option. See *note* box here

**Comment by chelauk** [comment]
Many thanks.
So autoMounts is set. I am using the nf-core template 
   profiles {
    singularity {
        singularity.enabled    = true
        singularity.autoMounts = true
        docker.enabled         = false
        podman.enabled         = false
        shifter.enabled        = false
        charliecloud.enabled   = false
    }   
}
but what I noticed was that when I launched nextflow using the full path, singularity successfully run the scripts in the bin folder of projectDir
so 
nextflow run  /full/path/projectDir   -profile singularity succeeds in running scripts projectDir/bin
while
nextflow run /path/with/soft/link/to/projectDir  -profile singularity
 nextflow runs but singularity fails to run scripts in the projectDir/bin folder.  I was wondering if this may be of some interest
